packages/jss/src/index.d.ts

```diff
@@ -1,35 +1,156 @@
+}
+
+interface RuleOptions {
+  selector?: string
+  sheet?: StyleSheet
+  index?: number
+  parent?: ContainerRule | StyleSheet
+  classes: Classes
+  jss: Jss
+  generateId: GenerateId
+  Renderer: Renderer
+}
+
+declare class BaseRule {
+  type: string
+  key: string
+  isProcessed: boolean
+  // eslint-disable-next-line no-use-before-define
+  options: RuleOptions
+  constructor(key: string, style: Style, options: RuleOptions)
+  toString(options?: ToCssOptions): string
+}
+
+interface ContainerRule extends BaseRule {
+  rules: RuleList
+}
+
+export interface Plugin {
+  onCreateRule?(name: string, decl: Style, options: RuleOptions): Rule
+  onProcessRule?(rule: Rule, sheet?: StyleSheet): void
+  onProcessStyle?(style: Style, rule: Rule, sheet?: StyleSheet): Style
+  onProcessSheet?(sheet?: StyleSheet): void
+  onChangeValue?(value: string, prop: string, rule: Rule): string | null | false
+  onUpdate?(data: object, rule: Rule, sheet?: StyleSheet): void
 }
 
-export interface Rule {
-  className: string;
-  selector: string;
-  applyTo(element: HTMLElement): void;
-  prop(key: string): string;
-  prop(key: string, value: any): this;
-  toJSON(): string;
+export type Rule = BaseRule | ContainerRule
+
+export declare class Renderer {
+  constructor(sheet?: StyleSheet)
+  setProperty(cssRule: HTMLElement | CSSStyleRule, prop: string, value: JssValue): boolean
+  getPropertyValue(cssRule: HTMLElement | CSSStyleRule, prop: string): string
+  removeProperty(cssRule: HTMLElement | CSSStyleRule, prop: string): void
+  setSelector(cssRule: CSSStyleRule, selectorText: string): boolean
+  attach(): void
+  detach(): void
+  deploy(sheet: StyleSheet): void
+  insertRule(rule: Rule): false | CSSRule
+  deleteRule(cssRule: CSSRule): boolean
+  replaceRule(cssRule: CSSRule, rule: Rule): false | CSSRule
+  indexOf(cssRule: CSSRule): number
+  getRules(): CSSRuleList | void
+}
+
+interface RuleFactoryOptions {
+  selector?: string
+  classes?: object
+  sheet?: StyleSheet
+  index?: number
+  jss?: Jss
+  generateId?: GenerateId
+  Renderer?: Renderer
+}
+
+export interface StyleSheetFactoryOptions {
+  media?: string
+  meta?: string
+  index?: number
+  link?: boolean
+  element?: HTMLStyleElement
+  generateId?: GenerateId
+  classNamePrefix?: string
+}
+
+interface StyleSheetOptions extends StyleSheetFactoryOptions {
+  index: number
+  generateId: GenerateId
+  Renderer: Renderer
+  insertionPoint?: InsertionPoint
+  jss: Jss
+}
+
+declare class SheetsRegistry {
+  readonly index: number
+  add(sheet: StyleSheet): void
+  reset(): void
+  remove(sheet: StyleSheet): void
+  toString(options?: ToCssOptions): string
+}
+
+declare class SheetsManager {
+  readonly size: number
+  get(key: object): StyleSheet | null
+  add(key: object, sheet: StyleSheet): void
+  manage(key: object): StyleSheet | null
+  unmanage(key: object): void
 }
 
-export interface StyleSheet<RuleName extends string = any> {
+export class StyleSheet<RuleName extends string = string> {
```

Comment:
I don't see a publicly exported class named `StyleSheet`, so changing this to `class` is incorrect.